core/model/context.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package model
+
+import "context"
+
+type contextKey string
+
+const (
+	// ContextKeyModelUUID is the key used to store the model UUID in the
+	// context.
+	ContextKeyModelUUID contextKey = "model-uuid"
+)
+
+// WitContextModelUUID returns a new context with the model UUID set.
+func WithContextModelUUID(ctx context.Context, modelUUID UUID) context.Context {
```

Comment:
Not saying no, just saying this reads funny to me.

`WithModelUUIDContext` sounds a little better. Up to you if, just writing words at this stage.

Reply:
This follows a common go'ism. I'll leave for now because if we want to change this, we should change all of them to be consistent.